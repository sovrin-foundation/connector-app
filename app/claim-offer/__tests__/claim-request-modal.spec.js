// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ClaimRequestStatusModal } from '../claim-request-modal'
import { CLAIM_REQUEST_STATUS } from '../type-claim-offer'
import type { ClaimRequestStatusModalProps } from '../type-claim-offer'
import {
  claimOfferPayload,
  senderLogoUrl,
  getStore,
} from '../../../__mocks__/static-data'

describe('<ClaimRequestStatusModal />', () => {
  const store = getStore()

  function getProps(props) {
    return {
      onContinue: jest.fn(),
      payload: claimOfferPayload,
      senderLogoUrl,
      claimRequestStatus: CLAIM_REQUEST_STATUS.SEND_CLAIM_REQUEST_SUCCESS,
      onModalHide: jest.fn(),
      ...props,
    }
  }

  function getWrapper(props) {
    return renderer.create(
      <Provider store={store}>
        <ClaimRequestStatusModal {...props} />
      </Provider>
    )
  }

  function setup(extraProps) {
    const props = getProps(extraProps)
    const wrapper = getWrapper(props)
    const instance = wrapper.root.findByType(ClaimRequestStatusModal).instance

    return { props, wrapper, instance }
  }

  function getPaidClaimProps() {
    return {
      payTokenValue: '1000',
      message1: 'You paid',
      message3: '',
      message6: 'They will issue it to you shortly.',
    }
  }

  it('should show check mark, you accepted text, if claim request is sent', () => {
    const message = {
      message1: 'You accepted',
      message3: 'from',
      message5: '!',
    }
    const { wrapper } = setup(message)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should match snapshot when pending popOver', () => {
    const extraProps = {
      message1: 'As soon as ',
      message3: ' signs and issues ',
      message5: ' to you it will appear here',
      isPending: true,
    }
    const { wrapper } = setup(extraProps)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should show arrow right and only Paying text if paid credential request is being sent', () => {
    const paidClaimProps = getPaidClaimProps()
    const extraProps = {
      claimRequestStatus: CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST,
      isPending: true,
      ...paidClaimProps,
    }
    const { wrapper } = setup(extraProps)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should show check mark, token amount, if paid credential request is sent', () => {
    const paidClaimProps = getPaidClaimProps()
    const extraProps = {
      claimRequestStatus: CLAIM_REQUEST_STATUS.PAID_CREDENTIAL_REQUEST_SUCCESS,
      ...paidClaimProps,
    }
    const { wrapper } = setup(extraProps)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should show arrow right and accepting text, if claim request is being sent', () => {
    const extraProps = {
      message1: 'You accepted',
      message3: 'from',
      message6: '',
      claimRequestStatus: CLAIM_REQUEST_STATUS.SENDING_CLAIM_REQUEST,
    }
    const { wrapper } = setup(extraProps)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should hideModal if continue is clicked', () => {
    const extraProps = {
      message1: 'You accepted ',
      message3: 'from',
      message5: '!',
    }
    const { wrapper, instance } = setup(extraProps)
    instance.onContinue()
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should match paid credential success, if sending state changes to success', () => {
    const paidClaimProps = getPaidClaimProps()
    const extraProps = {
      claimRequestStatus: CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST,
      isPending: true,
      ...paidClaimProps,
    }
    const { wrapper, props } = setup(extraProps)
    // update component so that state changes from sending paid credential
    // to paid credential success
    wrapper.update(
      <Provider store={store}>
        <ClaimRequestStatusModal
          {...{
            ...props,
            isPending: false,
            claimRequestStatus:
              CLAIM_REQUEST_STATUS.PAID_CREDENTIAL_REQUEST_SUCCESS,
          }}
        />
      </Provider>
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
