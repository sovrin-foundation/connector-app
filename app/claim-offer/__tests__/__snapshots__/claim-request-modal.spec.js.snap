// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimRequestStatusModal /> should hideModal if continue is clicked 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    shadow={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 10,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessibilityLabel="claim-request-message"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A8FE3",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "color": "#85BF43",
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should match paid credential success, if sending state changes to success 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    shadow={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 10,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessibilityLabel="claim-request-message"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Paying...
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A8FE3",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "color": "#85BF43",
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should match snapshot when pending popOver 1`] = `
<View
  center={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    center={true}
    fifth={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "height": "84%",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
      >
        
      </Text>
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Image
          collapsable={undefined}
          source={1}
          style={
            Object {
              "height": 30,
              "tintColor": "#777777",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 30,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#464646",
              },
              Object {
                "fontSize": 17,
                "fontWeight": "500",
                "letterSpacing": -0.42,
                "lineHeight": 21,
                "marginTop": 20,
              },
            ]
          }
        >
          Please wait…
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should show arrow right and accepting text, if claim request is being sent 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    shadow={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 10,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessibilityLabel="claim-request-message"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        Accepting...
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A8FE3",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "color": "#85BF43",
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should show arrow right and only Paying text if paid credential request is being sent 1`] = `
<View
  center={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    center={true}
    fifth={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "height": "84%",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
      >
        Paying...
      </Text>
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Image
          collapsable={undefined}
          source={1}
          style={
            Object {
              "height": 30,
              "tintColor": "#777777",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 30,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#464646",
              },
              Object {
                "fontSize": 17,
                "fontWeight": "500",
                "letterSpacing": -0.42,
                "lineHeight": 21,
                "marginTop": 20,
              },
            ]
          }
        >
          Please wait…
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should show check mark, token amount, if paid credential request is sent 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    shadow={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 10,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessibilityLabel="claim-request-message"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A8FE3",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "color": "#85BF43",
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ClaimRequestStatusModal /> should show check mark, you accepted text, if claim request is sent 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    shadow={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 10,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "shadowColor": "rgba(0, 0, 0, 0.2)",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 30,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3f3f3f",
          "borderBottomWidth": 0.2,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 8,
        }
      }
      verticalSpace={true}
    >
      <Text
        accessibilityLabel="claim-request-message"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "marginBottom": 5,
            },
          ]
        }
        testID="claim-request-message"
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A8FE3",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "color": "#85BF43",
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
